cli/cmd/check.go

```diff
@@ -254,103 +255,75 @@ func runExtensionChecks(cmd *cobra.Command, wout io.Writer, werr io.Writer, opts
+		switch extension {
 		case jaegerCmd.JaegerExtensionName:
-			jaegerCheckCmd := jaegerCmd.NewCmdCheck()
-			err = executeExtensionCheck(cmd, "jaeger", jaegerCheckCmd.Flags())
+			path = os.Args[0]
+			args = append([]string{"jaeger"}, args...)
 		case vizHealthCheck.VizExtensionName:
-			vizCmd := vizCmd.NewCmdCheck()
-			err = executeExtensionCheck(cmd, "viz", vizCmd.Flags())
+			path = os.Args[0]
+			args = append([]string{"viz"}, args...)
 		case mcCmd.MulticlusterExtensionName:
-			mcCheckCmd := mcCmd.NewCmdCheck()
-			err = executeExtensionCheck(cmd, "multicluster", mcCheckCmd.Flags())
+			path = os.Args[0]
+			args = append([]string{"multicluster"}, args...)
 		default:
-			// Since we don't have checks for extensions we don't support
-			// create a healthchecker that checks if the namespace exists
-			categoryID := healthcheck.CategoryID(ns.Labels[k8s.LinkerdExtensionLabel])
-			checkers := []healthcheck.Checker{}
-			checkers = append(checkers,
-				*healthcheck.NewChecker(fmt.Sprintf("%s extension Namespace exists", categoryID)).
-					WithHintAnchor(fmt.Sprintf("%s-ns-exists", categoryID)).
-					Fatal().
-					WithCheck(func(ctx context.Context) error {
-						_, err := kubeAPI.GetNamespaceWithExtensionLabel(ctx, string(categoryID))
-						if err != nil {
-							return err
-						}
-						return nil
-					}))
-			hc := healthcheck.NewHealthChecker([]healthcheck.CategoryID{categoryID},
-				&healthcheck.Options{
-					ControlPlaneNamespace: controlPlaneNamespace,
-					CNINamespace:          cniNamespace,
-					DataPlaneNamespace:    opts.namespace,
-					KubeConfig:            kubeconfigPath,
-					KubeContext:           kubeContext,
-					Impersonate:           impersonate,
-					ImpersonateGroup:      impersonateGroup,
-					APIAddr:               apiAddr,
-					VersionOverride:       opts.versionOverride,
-					RetryDeadline:         time.Now().Add(opts.wait),
-					CNIEnabled:            opts.cniEnabled,
-				})
-			hc.AppendCategories(*healthcheck.NewCategory(categoryID, checkers, true))
-			success := healthcheck.RunChecks(wout, werr, hc, opts.output)
-			if !success {
-				os.Exit(1)
+			extensionCmd := fmt.Sprintf("linkerd-%s", extension)
+			path, err = exec.LookPath(extensionCmd)
+			results.Results = []healthcheck.CheckResult{
+				{
+					Category:    healthcheck.CategoryID(extensionCmd),
+					Description: fmt.Sprintf("Linkerd extension command %s exists", extensionCmd),
+					Err:         err,
+					HintAnchor:  "extensions",
+					Warning:     true,
+				},
 			}
 		}
-		if err != nil {
-			return err
+		if err == nil {
+			plugin := exec.Command(path, args...)
+			output, _ := plugin.Output()
+			extensionResults, err := healthcheck.ParseJSONCheckOutput(output)
+			if err != nil {
+				command := fmt.Sprintf("%s %s", path, strings.Join(args, " "))
+				return false, fmt.Errorf("invalid extension check output from \"%s\" (JSON object expected):\n%s\n[%s]", command, string(output), err)
```

Comment:
If I run for example `linkerd check -L foobar` having linkerd installed in its default namespace, the core checks will fail as expected, and so will the multicluster checks with a message like:
```
failed to run extensions checks: invalid extension check output from "/home/alpeb/src/pr/5762/linkerd2/target/cli/linux-amd64/linkerd multicluster check --as-group=[] --linkerd-namespace=foobar --verbose=false --proxy=false --wait=5m0s --output=json" (JSON object expected):
```
The error message returned from multicluster won't be shown because we're not capturing stderr from the `plugin.Output()` call.
Also I think it'd be better here to not short-circuit and instead output the error message, set success to false, and continue with the other extensions.

Reply:
Good point.  I've updated to handle this more gracefully by reporting the stderr if it exists, and the json parse error otherwise, and by not short circuiting in the case of an error:

```
linkerd-existence
-----------------
× 'linkerd-config' config map exists
    configmaps "linkerd-config" not found
    see https://linkerd.io/checks/#l5d-existence-linkerd-config for hints

Status check results are ×

Linkerd extensions checks
=========================

linkerd-foo
-----------
√ Linkerd extension command linkerd-foo exists
√ everything looks ship shape, captain!
‼ actually, hold on...
    something is fishy here...
× oh no!
    abandon ship!

Status check results are ×

linkerd-linkerd-multicluster
----------------------------
× Running: /Users/alexanderleong/workspace/linkerd2/target/cli/darwin/linkerd multicluster check --as-group=[] --linkerd-namespace=foobar --verbose=false --proxy=false --wait=5m0s --output=json
    Failed to fetch linkerd config: configmaps "linkerd-config" not found

    see https://linkerd.io/checks/#extensions for hints

Status check results are ×

linkerd-viz
-----------
√ linkerd-viz Namespace exists
√ linkerd-viz ClusterRoles exist
√ linkerd-viz ClusterRoleBindings exist
√ tap API server has valid cert
√ tap API server cert is valid for at least 60 days
√ tap API service is running
√ linkerd-viz pods are injected
√ viz extension pods are running
√ prometheus is installed and configured correctly
√ can initialize the client
√ viz extension self-check

Status check results are √
```

Reply:
Specifically when reporting this kind of error, we report the category as `linkerd-linkerd-multicluster`.  Rather than adding special handling here to fix this, we should (in a separate PR) change the `linkerd.io/extension` label of the built-in extensions to remove the linkerd prefix.  Once that is changed, this will correctly report as `linkerd-multicluster`.